urllib3/sessionmanager.py

```diff
@@ -0,0 +1,77 @@
+from .util.retry import Retry
+from .request import RequestMethods, Request
+from .util.sessioncontext import SessionContext
+
+
+class SessionManager(RequestMethods):
+    """
+    Allows arbitrary requests while maintaining session context across
+    those requests. Currently, that context consists of automatic
+    cookie storage and retrieval.
+
+    :param manager:
+        An appropriate :class:`urllib3.poolmanager.PoolManager` or
+        :class:`urllib3.poolmanager.ProxyManager` object
+        to handle HTTP requests for the SessionManager
+
+    :param context:
+        A predefined :class:`urllib3.util.context.SessionContext` object to use in the session;
+        if not provided, a new one will be created.
+
+    :param headers:
+        Headers to include with all requests, unless other
+        headers are given explicitly.
+
+    Example::
+
+        >>> manager = SessionManager(PoolManager())
```

Comment:
I have mixed feelings about this. I feel like if this is how most people are going to use it, we shouldn't require it be provided.